prow/pod-utils/decorate/podspec.go

```diff
@@ -593,6 +595,22 @@ func decorate(spec *coreapi.PodSpec, pj *prowapi.ProwJob, rawEnv map[string]stri
 	// TODO(fejta): we should pass around volume names rather than forcing particular mount paths.
 
 	rawEnv[artifactsEnv] = artifactsPath
+
+	if gcsConfig := pj.Spec.DecorationConfig.GCSConfiguration; gcsConfig != nil {
+		jobSpec := downwardapi.NewJobSpec(pj.Spec, rawEnv[downwardapi.BuildIDEnv], pj.Name)
+		_, gcsPath, _ := gcsupload.PathsForJob(gcsConfig, &jobSpec, "")
+
+		u, err := url.Parse(gcsConfig.Bucket)
+		if err != nil {
+			return fmt.Errorf("could not parse the GCS bucket name %s: %v", gcsConfig.Bucket, err)
+		}
+		if u.Scheme == "" {
+			u.Scheme = "gs"
```

Comment:
Please include unit tests for other schemes. In particular, the cases listed here: https://github.com/kubernetes/test-infra/pull/17215#discussion_r410331907